packages/web3torrent/src/components/file-info/progress-bar/ProgressBar.tsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import "./ProgressBar.scss";
+
+export type ProgressBarProps = {
+  downloaded: number;
+  size: number;
+  status: "Completed" | "Downloading";
+  stopAction?: () => void;
+};
+
+export const ProgressBar: React.FC<ProgressBarProps> = ({
+  downloaded,
+  size,
+  status = "Downloading"
+}: ProgressBarProps) => {
+  return (
+    <div className="progress-bar" data-label={downloaded + "/" + size}>
+      <span className="filler" style={{ width: `${(downloaded / size) * 100}%` }} />
+      <span className="bar-progress">
+        {downloaded}Mb/{size}Mb
+      </span>
+      <span className="bar-status">{status}</span>
+      <button type="button" className="bar-cancelButton">
+        X
```

Comment:
![](https://media.giphy.com/media/jIRPOnUASNsQw/giphy.gif)

Reply:
(In all seriousness, feel free to ~copypaste~ get inspired from here: https://github.com/statechannels/monorepo/blob/master/packages/embedded-wallet/src/ui/dialog/Dialog.module.css#L25-L63)

Reply:
Didn't use that (couldn't make it work, I believe because I had absolute positioning in the element), but replaced the X with a SVG element